pkg/client/create_manifest.go

```diff
@@ -2,60 +2,83 @@ package client
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/buildpacks/imgutil"
-	"github.com/pkg/errors"
+	"github.com/buildpacks/imgutil/index"
+	ggcrName "github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/types"
 )
 
 type CreateManifestOptions struct {
-	// Name of the ManifestList.
-	ManifestName string
-
-	// List of Images
-	Manifests []string
-
-	// Manifest list type (oci or v2s2) to use when pushing the list (default is v2s2)
-	MediaType imgutil.MediaTypes
-
-	// Skip creating index locally, directly publish to a registry.
-	// Requires ManifestName to be a valid registry location.
-	Publish bool
+	Format, Registry       string
+	Insecure, Publish, All bool
+}
 
-	// Defines the registry to publish the manifest list.
-	Registry string
+// CreateManifest implements commands.PackClient.
+func (c *Client) CreateManifest(ctx context.Context, name string, images []string, opts CreateManifestOptions) (err error) {
+	ops := parseOptsToIndexOptions(opts)
+	_, err = c.indexFactory.LoadIndex(name, ops...)
+	if err == nil {
+		return fmt.Errorf("image index with name: '%s' exists", name)
```

Comment:
I will add `exits in your local storage, use 'pack manifest remove' if you want to delete it`